src/nodes.coffee

```diff
@@ -1301,20 +1301,19 @@ exports.Class = class Class extends Base
     @hasNameClash = @name? and @name is parentName
 
     if executableBody or @hasNameClash
-      @compileNode = @compileClassDeclaration
-      result = new ExecutableClassBody(@, executableBody).compileToFragments o
-      @compileNode = @constructor::compileNode
+      node = new ExecutableClassBody @, executableBody
     else
-      result = @compileClassDeclaration o
-
-      # Anonymous classes are only valid in expressions
-      result = @wrapInParentheses result if not @name? and o.level is LEVEL_TOP
+      node = @
+      node = new Parens node if not @name? and o.level is LEVEL_TOP
 
     if @variable
-      assign = new Assign @variable, new Literal(''), null, { @moduleDeclaration }
-      [ assign.compileToFragments(o)..., result... ]
-    else
-      result
+      node = new Assign @variable, node, null, { @moduleDeclaration }
+
+    @compileNode = @compileClassDeclaration
+    try
+      return node.compileToFragments o
```

Comment:
It's possible that we should have a separate `ClassDeclaration` node, which would avoid these shenanigans. That feels like a bit too much of a refactor for this issue though.